### PR TITLE
Add Restart File Output for Fluid-In-Place Arrays

### PR DESCRIPTION
**Pull request by bska**
This PR introduces special case logic for outputting per-cell fluid-in-place arrays.  In particular, we intercept arrays tagged as `RESTART_SOLUTION` and named as

  * FIP{OIL,WAT,GAS}
  * RFIP{OIL,WAT,GAS}
  * SFIP{OIL,WAT,GAS}

and write those in a separate function (`writeFluidInPlace()`) instead of as part of `writeRegularSolutionVectors()`.  We also add special compatibility logic to output `SFIP*` as alias/copy of `FIP*` if no `RFIP*` or `SFIP*` arrays exist in [`RestartValue::solution`](https://github.com/OPM/opm-common/blob/c4d25b479fb810da828ca0cb033a1dc2e4434055/opm/output/eclipse/RestartValue.hpp#L78).

### CODE REVIEW

**Comment by bska** [comment]
I'm creating this PR in draft mode because it depends on, and contains, the upstream PR #3951.  We also need the companion PR OPM/opm-simulators#5224 because that PR will create the requisite output arrays on the simulator side.

**Comment by bska** [comment]
> jenkins build this opm-simulators=5224 please

The [test failures](https://ci.opm-project.org/job/opm-common-PR-builder/6799/testReport/) are all expected due to now outputting additional arrays to the restart file.  When we're ready to review/merge this I'll rerun the build check using `ignore_extra` to verify.

**Comment by bska** [comment]
jenkins build this opm-simulators=5224 ignore_extra please

**Comment by bska** [comment]
I'm marking this as "ready for review" now.  The PR introduces a couple of new arrays in certain regression tests and we must update the reference solutions for those.  This PR must be merged together with the downstream PR OPM/opm-simulators#5224.

**Comment by bska** [comment]
jenkins build this opm-simulators=5224 ignore_extra please

File: opm/output/eclipse/RestartIO.cpp [hunk 1]
```diff
@@ -654,6 +680,64 @@ namespace {
...
+
+        if (anyRSFip) {
+            // The simulator provides properly tagged in-place arrays.  No
+            // further action needed.
+            return;
+        }
+
+        // If we get here, all fluid-in-place vectors have the name FIP* and
+        // represent surface condition volumes.  Output the same vectors
+        // using the correaponding SFIP name as well.
```
Comment (akva2) [inline opm/output/eclipse/RestartIO.cpp:735]: corresponding

**Review by akva2 — APPROVED** [review]
Looks good, one typo located.

File: opm/output/eclipse/RestartIO.cpp (hunk 1 above)
Comment (bska) [reply opm/output/eclipse/RestartIO.cpp]: > corresponding

Thanks.  I've pushed an update to fix that.

**Comment by bska** [comment]
jenkins build this opm-simulators=5224 update_data please

**Comment by akva2** [comment]
jenkins build this opm-simulators=5224 update_data please

**Comment by bska** [comment]
jenkins build this opm-simulators=5224 update_data please

**Comment by bska** [comment]
jenkins build this opm-simulators=5224 opm-tests=1135 please

**Comment by bska** [comment]
Build check is green, the PR's approved, and the new reference solutions have been installed on the CI system.  I'm merging this into the master branch now.

**Pull request merged**